### PR TITLE
Diagnostics fixup

### PR DESCRIPTION
**Pull request by laughedelic**
After #204 I observed that you cannot switch `scalac.diagnostics.enabled` and `scalafix.enabled` independently. So I modified `SquiggliesProvider` (and renamed it to `DiagnosticsProvider`) to work equally with the diagnostics from scalac and from scalafix. 

I tried also to get some diagnostics from `ScalacErrorReporter`, but no matter what I did, I've never seen them, so I assumed that it actually doesn't work. And it doesn't seem necessary now. Let me know if I'm wrong.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/metals/ScalacErrorReporter.scala [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-  }
-
-  private def toSeverity(
-      s: semanticdb.Severity
-  ): l.DiagnosticSeverity =
-    s match {
-      case semanticdb.Severity.Error => l.DiagnosticSeverity.Error
-      case semanticdb.Severity.Warning => l.DiagnosticSeverity.Warning
-      case semanticdb.Severity.Info => l.DiagnosticSeverity.Information
-    }
```
Comment (laughedelic) [inline metals/src/main/scala/scala/meta/metals/ScalacErrorReporter.scala]: Conversion methods here duplicated the code from `ScalametaEnrichments`, so the only valuable code here was in the `analyzeIndex` methods

File: metals/src/main/scala/scala/meta/metals/providers/DiagnosticsProvider.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    diagnostics(m.Database(doc :: Nil))
+  def diagnostics(db: m.Database): Task[Seq[PublishDiagnostics]] = {
+    if (scalafixDisabled && scalacDisabled) Task(Nil)
+    else Task.sequence {
+      db.documents.map { document =>
+        val uri = document.input.syntax
+
+        val scalacDiagnostics: Seq[Diagnostic] =
+          if (scalacDisabled) Nil
+          else document.messages.map(_.toLSP("scalac"))
```
Comment (laughedelic) [inline metals/src/main/scala/scala/meta/metals/providers/DiagnosticsProvider.scala]: I was doubting whether this should use the code from `ScalaErrorReporter` instead, like this:

```scala
  private def scalacAnalyze(doc: m.Document): Seq[Diagnostic] = {
    val index = EagerInMemorySemanticdbIndex(
      m.Database(doc :: Nil),
      m.Sourcepath(Nil),
      m.Classpath(Nil)
    )
    for {
      doc <- index.database.documents
      msg <- doc.messages
    } yield msg.toLSP("scalac")
  }
```

I tried it, but it was glitchy so I left it as is. I don't understand very well why `EagerInMemorySemanticdbIndex` was used there and whether it should be used here. Semanticdbs we get here are already built interactively, isn't it enough?

File: metals/src/main/scala/scala/meta/metals/providers/DiagnosticsProvider.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  def diagnostics(db: m.Database): Task[Seq[PublishDiagnostics]] = {
+    if (scalafixDisabled && scalacDisabled) Task(Nil)
+    else
+      Task.sequence {
+        db.documents.map { document =>
+          val uri = document.input.syntax
+
+          val scalacDiagnostics: Seq[Diagnostic] =
+            if (scalacDisabled) Nil
+            else document.messages.map(_.toLSP("scalac"))
```
Comment (laughedelic) [inline metals/src/main/scala/scala/meta/metals/providers/DiagnosticsProvider.scala]: I was doubting whether this should use the code from `ScalaErrorReporter` instead, like this:

```scala
  private def scalacAnalyze(doc: m.Document): Seq[Diagnostic] = {
    val index = EagerInMemorySemanticdbIndex(
      m.Database(doc :: Nil),
      m.Sourcepath(Nil),
      m.Classpath(Nil)
    )
    for {
      doc <- index.database.documents
      msg <- doc.messages
    } yield msg.toLSP("scalac")
  }
```

I tried it, but it was glitchy so I left it as is. I don't understand very well why `EagerInMemorySemanticdbIndex` was used there and whether it should be used here. Semanticdbs we get here are already built interactively, isn't it enough?

File: metals/src/main/scala/scala/meta/metals/ScalacErrorReporter.scala (hunk 1 above)
Comment (laughedelic) [inline metals/src/main/scala/scala/meta/metals/ScalacErrorReporter.scala]: Conversion methods here duplicated the code from `ScalametaEnrichments`, so the only valuable code here was in the `analyzeIndex` methods

**Comment by laughedelic** [comment]
@olafurpg thanks for testing. 

Thinking about it again, I found a bit confusing situation: 
* `scalac.completions.enabled = true` (=> `scalac.enabled = true`)
* `scalac.diagnostics.enabled = false` 
* `scalafix.enabled = true`

Right now it means that interactive semanticdbs will be produced, because completions are on, but scalac diagnostics will be still filtered out in the `DiagnosticsProvider`, while scalafix ones no. Isn't it a bit strange? On one hand it's what the user wants, on the other hand those diagnostics are already there, just not shown.

**Comment by olafurpg** [comment]
> Isn't it a bit strange? 

Not at all, this is in fact a setup I would like to use personally. Completions have a much higher tolerance for false positives than diagnostics. I would prefer to get diagnostics from the build instead of the presentation compiler.

**Pull request merged**